chore: Semantic release

sunnieeeYoon opened the pull request:
<!--
  Thank you for opening a pull request for JunbaeJs.
-->

### Description of change
- applying semantic-release
- adding 'breaking' and 'ci' to Commit Convention.
- applying github action
<img width="805" alt="스크린샷 2024-01-31 오후 4 33 57" src="https://github.com/JunbaeJs/junbae-mode/assets/152782031/e8b72228-d16e-4ae6-9ad5-6dc48a749fc7">


<!--
  Please be clear and concise what the change is intended to do,
  why this change is needed, and how you've verified that it
  corrects what you intended.

  In some cases it may be helpful to include the current behavior
  and the new behavior.
-->

### Pull-Request Checklist

<!--
  Please make sure to review and check all of the following.

  If an item is not applicable, you can add "N/A" to the end.
-->

- [x] Code is up-to-date with the `main` branch
- [ ] There are new or updated unit tests validating the change
- [x] Documentation has been updated to reflect this change
- [x] The new commits follow conventions explained in [CONTRIBUTING.md](https://github.com/JunbaeJs/junbae-mode/blob/main/CONTRIBUTING.md)

<!--
  🎉 Thank you for contributing and making JunbaeJs even better!
-->

Commit:
ci: semantic-release

Commit:
docs: edit commit convention

changchanghwang commented on package.json:
```diff
@@ -58,5 +59,147 @@
+  "release": {
+    "branches": [
+      {
+        "name": "master"
```
```suggestion
        "name": "main"
```

changchanghwang commented on package.json:
```diff
@@ -58,5 +59,147 @@
+        "name": "master"
+      },
+      {
+        "name": "dev",
+        "prerelease": "dev"
```
```suggestion
        "name": "develop",
        "prerelease": "develop"
```

changchanghwang commented on CONTRIBUTING.md:
```diff
@@ -3,11 +3,13 @@
 refactor: refactoring production code, eg. renaming a variable
 test: adding missing tests, refactoring tests; no production code change
 chore: updating grunt tasks etc; no production code change
+ci: ci update
```
We don't need this. Please use `chore:`

changchanghwang commented on package.json:
```diff
@@ -58,5 +59,147 @@
+                "section": "📦 Chores",
+                "hidden": true
+              },
+              {
+                "type": "ci",
+                "section": "💫 CI/CD",
+                "hidden": false
+              },
```
We don't need this - 2.

changchanghwang commented on package.json:
```diff
@@ -58,5 +59,147 @@
+                "type": "test",
+                "section": "✅ Tests",
+                "hidden": true
+              }
```
add breaking plz

changchanghwang commented on package.json:
```diff
@@ -58,5 +59,147 @@
       }
     ]
+  },
+  "release": {
```
Please move this all changes to `.releaserc`

sunnieeeYoon committed:
chore: add ci & reflect comments

sunnieeeYoon committed:
fix: test

sunnieeeYoon committed:
fix: test

sunnieeeYoon committed:
fix: add semantic-release package

semantic-release-bot committed:
chore(release): 1.0.0 [skip ci]

# 1.0.0 (2024-01-31)

### Bug Fixes

* add semantic-release package ([14800ae](https://github.com/JunbaeJs/junbae-mode/commit/14800aeb080e0ed58a480b9f12a145a0941e805c))
* mistake ([c697174](https://github.com/JunbaeJs/junbae-mode/commit/c697174ef1760c48ec5a67ec22f586aa98403adc))
* out -> dist ([dfc08b9](https://github.com/JunbaeJs/junbae-mode/commit/dfc08b912cca310d994bfe443dcaad47763b40fd))
* test ([1f383d7](https://github.com/JunbaeJs/junbae-mode/commit/1f383d78f27075484053b884627fb044ba09c0df))
* test ([a59a6ce](https://github.com/JunbaeJs/junbae-mode/commit/a59a6ce836f921672c5d446cd16a5118370c27c6))

### Features

* initial pubslih to vscode ([bd2225e](https://github.com/JunbaeJs/junbae-mode/commit/bd2225e2952313d31e05a27f19798cc57e419530))

sunnieeeYoon committed:
fix: edit package.json

sunnieeeYoon committed:
Merge branch 'semantic-release' of https://github.com/JunbaeJs/junbae-mode into semantic-release

semantic-release-bot committed:
chore(release): 1.0.1 [skip ci]

## [1.0.1](https://github.com/JunbaeJs/junbae-mode/compare/v1.0.0...v1.0.1) (2024-01-31)

### Bug Fixes

* edit package.json ([def8e5a](https://github.com/JunbaeJs/junbae-mode/commit/def8e5a72fbac988d6ec9e13675fe3bade911f31))

sunnieeeYoon committed:
fix: test for releaserc

sunnieeeYoon committed:
Merge branch 'semantic-release' of https://github.com/JunbaeJs/junbae-mode into semantic-release

semantic-release-bot committed:
chore(release): 1.0.2 [skip ci]

## [1.0.2](https://github.com/JunbaeJs/junbae-mode/compare/v1.0.1...v1.0.2) (2024-01-31)

### Bug Fixes

* test for releaserc ([497fe0a](https://github.com/JunbaeJs/junbae-mode/commit/497fe0a4220fdc1f06dee854506a4ff78ecc0edf))

sunnieeeYoon committed:
chore: edit cicd.yml & releaserc

sunnieeeYoon committed:
Merge branch 'semantic-release' of https://github.com/JunbaeJs/junbae-mode into semantic-release

changchanghwang commented on .github/workflows/cicd.yml:
```diff
@@ -0,0 +1,49 @@
+
+    strategy:
+      matrix:
+        node-version: [20.11.x]
```
How about 20.x ?

sunnieeeYoon replied:
<img width="1038" alt="스크린샷 2024-02-01 오후 3 51 50" src="https://github.com/JunbaeJs/junbae-mode/assets/152782031/5d74ef56-ef69-4227-9cb2-dbe4ff538b01">
we require Node version 20.8.1 or higher. and LTS is 20.11.0

changchanghwang commented on .github/workflows/cicd.yml:
change File name plz. It Not include `cd`

changchanghwang commented on .github/workflows/cicd.yml:
```diff
@@ -0,0 +1,49 @@
+      - name: Release
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
```
I think it's not necessary for us. It means token for NPM registry

changchanghwang commented on .releaserc:
```diff
@@ -0,0 +1,139 @@
+        {
+          "preset": "angular",
+          "releaseRules": [
+            {
+              "scope": "no-release",
+              "release": false
+            },
```
What is this?

changchanghwang commented on .releaserc:
```diff
@@ -0,0 +1,139 @@
+              "type": "chore",
+              "release": false
+            },
+            {
+              "type": "ci",
+              "release": false
+            },
```
remove plz

changchanghwang commented on CHANGELOG.md:
?

changchanghwang commented on package.json:
```diff
@@ -1,7 +1,7 @@
   "name": "junbae-mode",
   "displayName": "Junbae Mode",
-  "version": "0.0.1",
+  "version": "1.0.2",
```
?

changchanghwang commented on package.json:
```diff
@@ -58,5 +59,11 @@
       }
     ]
+  },
+  "dependencies": {
+    "@semantic-release/changelog": "^6.0.3",
+    "@semantic-release/commit-analyzer": "^10.0.1",
+    "@semantic-release/git": "^10.0.1",
+    "@semantic-release/release-notes-generator": "^11.0.4"
   }
```
It's not dependencies. move to devDep

sunnieeeYoon committed:
chore: edit ci & package

changchanghwang reviewed (approved):
LGTM
